types/jsonwebtoken/v7/index.d.ts

```diff
@@ -0,0 +1,260 @@
+    error: Error | null,
+    encoded: string | undefined,
+) => void;
+
+// standard names https://www.rfc-editor.org/rfc/rfc7515.html#section-4.1
+export interface JwtHeader {
+    alg: string | Algorithm;
+    typ?: string | undefined;
+    cty?: string | undefined;
+    crit?: Array<string | Exclude<keyof JwtHeader, "crit">> | undefined;
+    kid?: string | undefined;
+    jku?: string | undefined;
+    x5u?: string | string[] | undefined;
+    "x5t#S256"?: string | undefined;
+    x5t?: string | undefined;
+    x5c?: string | string[] | undefined;
+}
+
+// standard claims https://datatracker.ietf.org/doc/html/rfc7519#section-4.1
+export interface JwtPayload {
+    [key: string]: any;
+    iss?: string | undefined;
+    sub?: string | undefined;
+    aud?: string | string[] | undefined;
+    exp?: number | undefined;
+    nbf?: number | undefined;
+    iat?: number | undefined;
+    jti?: string | undefined;
+}
+
+export interface Jwt {
+    header: JwtHeader;
+    payload: JwtPayload | string;
+    signature: string;
+}
+
+// https://github.com/auth0/node-jsonwebtoken#algorithms-supported
+export type Algorithm =
+    | "HS256"
+    | "HS384"
+    | "HS512"
+    | "RS256"
+    | "RS384"
+    | "RS512"
+    | "ES256"
+    | "ES384"
+    | "ES512"
+    | "PS256"
+    | "PS384"
+    | "PS512"
+    | "none";
+
+export type SigningKeyCallback = (
+    error: Error | null,
+    signingKey?: Secret,
+) => void;
+
+export type GetPublicKeyOrSecret = (
+    header: JwtHeader,
+    callback: SigningKeyCallback,
+) => void;
+
+export type Secret =
+    | string
+    | Buffer
+    | KeyObject
+    | { key: string | Buffer; passphrase: string };
+
+/**
+ * Synchronously sign the given payload into a JSON Web Token string
+ * payload - Payload to sign, could be an literal, buffer or string
+ * secretOrPrivateKey - Either the secret for HMAC algorithms, or the PEM encoded private key for RSA and ECDSA.
+ * [options] - Options for the signature
+ * returns - The JSON Web Token string
+ */
+export function sign(
+    payload: string | Buffer | object,
+    secretOrPrivateKey: Secret,
+    options?: SignOptions,
+): string;
+export function sign(
+    payload: string | Buffer | object,
+    secretOrPrivateKey: null,
+    options?: SignOptions & { algorithm: "none" },
+): string;
+
+/**
+ * Sign the given payload into a JSON Web Token string
+ * payload - Payload to sign, could be an literal, buffer or string
+ * secretOrPrivateKey - Either the secret for HMAC algorithms, or the PEM encoded private key for RSA and ECDSA.
+ * [options] - Options for the signature
+ * callback - Callback to get the encoded token on
+ */
+export function sign(
+    payload: string | Buffer | object,
+    secretOrPrivateKey: Secret,
+    callback: SignCallback,
+): void;
+export function sign(
+    payload: string | Buffer | object,
+    secretOrPrivateKey: Secret,
+    options: SignOptions,
+    callback: SignCallback,
+): void;
+export function sign(
```

Comment:
Without any specific knowledge of jsonwebtoken, the addition of these overloads seems like the riskiest part of the change. It's probably OK but I would like somebody who knows jsonwebtoken to take a look too.